### PR TITLE
feature(#10): ranges integration implemented

### PR DESCRIPTION
**Pull request by y9san9**
There are also changes to KBigNum and KBigInt, now they extend Comparable, since it is required to be generic upperbound in ClosedRange

### CODE REVIEW

**Comment by soywiz** [comment]
This code is probably outdated, because development happens on korge-next. Could you make the PR there? 

https://github.com/korlibs/korge-next

**Comment by razvizion** [comment]
This is problematic for some projects. For example we need watchOS ARM64 simulator targets. I was able to make whole korge-next work for ARM64 simulator (M1), bo no chance for watchOS, as all the packages should work with all declared targets. Imagine Korgw implementing obligatory interface for game window for watchOS target, it doesn't make sense. As for now I'm thinking about forking KBigNum and Krypto only to publish builds with all targets that we need. Even now on maven central Krypto artifacts for all watchOS targets are outdated 2.2.0.

**Comment by y9san9** [comment]
@soywiz Hi there! Do you mean that I should just move my code into `korge-next`? Or I need also to do some changes to the code? 

And why `korge-next` doesn't use git submodules for composing libraries?

**Comment by soywiz** [comment]
@y9san9 Sorry for the delay. What I meant is that the PR should be done in korge-next since the code is there, and this repo might not have all the changes that are already on korge-next and there might be conflicts

**Comment by soywiz** [comment]
So what I would propose is to bring those changes to: https://github.com/korlibs/korge-next/tree/master/kbignum/src/commonMain/kotlin/com/soywiz/kbignum

File: kbignum/src/commonMain/kotlin/com/soywiz/kbignum/ranges/ClosedBigNumRange.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package com.soywiz.kbignum.ranges
+
+import com.soywiz.kbignum.BigNum
+
+fun main() {
+    1f..2f
+}
```
Comment (soywiz) [inline kbignum/src/commonMain/kotlin/com/soywiz/kbignum/ranges/ClosedBigNumRange.kt:7]: I think this shouldn't be here

**Comment by soywiz** [comment]
@y9san9

I have moved back kbignum from korge-next to this repo since it is not used for the rest of the libraries. We can resume this PR here, but this might need some rebasing.

Also there is a `fun main` there that was probably commited by mistake

**Comment by soywiz** [comment]
Manually merged here: https://github.com/korlibs/kbignum/commit/04d6a99b7af85c0708926e5458648f8db1388be7

**Pull request closed without merging**